src/controls/Controls.spec.js

```diff
@@ -1 +1,30 @@
-// Test away!
\ No newline at end of file
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Controls from './Controls';
+import { render, fireEvent } from '@testing-library/react';
+
+describe('<Controls />', () => {
+
+    it('matches snapshot', () => {
+      const tree = renderer.create(<Controls />).toJSON();
+      expect(tree).toMatchSnapshot();
+    });
+
+    it('buttons toggle closed and locked', () => {
+      const { getByText } = render(<Controls />);
+      getByText (/Lock Gate/);
+      getByText (/Close Gate/);
```

Comment:
for tests, it's hard to write tests to semantically match what you say it's testing.  In this case, you don't really have any assertions.  What you do have are the queries.  However, the only thing you can assert with the `getBy` query is that there is exactly **1** _element_ with that matching text no more and no less.

To assert it's a button you'd need to query by role `getByRole('button')` and then furthermore `getNodeText` to assert the inside text.  

If you like you can also check the text before, then fire an event then check the text after to say it can toggle.